src/sentry/eventstream/kafka/postprocessworker.py

```diff
@@ -54,12 +57,33 @@ def _get_task_kwargs(message: Message) -> Optional[Mapping[str, Any]]:
             return get_task_kwargs_for_message(message.value())
 
 
+__metrics: MutableMapping[Tuple[int, str], int] = defaultdict(int)
+__metric_record_freq_sec = 1.0
+__last_flush = time.time()
+__lock = Lock()
+
+
 def _record_metrics(partition: int, task_kwargs: Mapping[str, Any]) -> None:
+    """
+    Records the number of messages processed per partition. Metric is flushed every second.
+    """
+    global __last_flush
     event_type = "transactions" if task_kwargs["group_id"] is None else "errors"
-    metrics.incr(
-        _MESSAGES_METRIC,
-        tags={"partition": partition, "type": event_type},
-    )
+    __metrics[(partition, event_type)] += 1
+
+    current_time = time.time()
+    if current_time - __last_flush > __metric_record_freq_sec:
+        with __lock:
+            metrics_to_send = __metrics
```

Comment:
Don't you need to copy the `__metrics` dict here?

Reply:
Oops, updated to redeclare the variable instead of using .clear()